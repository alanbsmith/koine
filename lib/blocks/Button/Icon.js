import React from 'react';
import styled from 'styled-components';
import FontAwesome from 'react-fontawesome';
import { omit } from 'lodash';
import { rem } from 'polished';
import { applyStyleModifiers } from 'styled-components-modifiers';

const modifierConfig = {
  left: () => `
    padding-right: ${rem(8)};
  `,
  right: () => `
    padding-left: ${rem(8)};
  `,
};

function UnstyledFontAwesome(props) {
  const iconProps = omit(props, ['modifiers']);
  return <FontAwesome {...iconProps} />;
}

const Icon = styled(UnstyledFontAwesome)`
  ${applyStyleModifiers(modifierConfig)}
`;

export default Icon;
