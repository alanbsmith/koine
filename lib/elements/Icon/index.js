import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import FontAwesome from 'react-fontawesome';
import { omit } from 'lodash';
import { rem } from 'polished';


function UnstyledIcon(props) {
  const iconProps = omit(props, ['modifiers']);
  return <FontAwesome {...iconProps} />;
}

const Icon = styled(UnstyledIcon)`
  font-size: ${props => rem(props.theme.dimensions.baseFontSize)};
`;

Icon.propTypes = {
  name: PropTypes.string.isRequired,
};

export default Icon;
