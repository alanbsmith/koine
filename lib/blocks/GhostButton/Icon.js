import styled from 'styled-components';
import { rem } from 'polished';
import { applyStyleModifiers } from 'styled-components-modifiers';

import { Icon as BasicIcon } from '../../elements';

const modifierConfig = {
  left: (props) => `
    padding-right: ${rem(props.theme.dimensions.basePad)};
  `,
  right: (props) => `
    padding-left: ${rem(props.theme.dimensions.basePad)};
  `,
};

const Icon = styled(BasicIcon)`
  ${applyStyleModifiers(modifierConfig)}
`;

export default Icon;
