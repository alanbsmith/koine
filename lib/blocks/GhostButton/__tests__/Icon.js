import React from 'react';
import Icon from '../Icon';

import { renderWithTheme } from '__tests__/helpers';


describe('Icon Element', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(
      <Icon name="home" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies left modifier', () => {
    const tree = renderWithTheme(
      <Icon
        modifiers={['left']}
        name="home"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies right modifier', () => {
    const tree = renderWithTheme(
      <Icon
        modifiers={['right']}
        name="home"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
