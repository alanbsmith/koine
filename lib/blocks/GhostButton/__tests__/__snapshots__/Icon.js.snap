// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Element applies left modifier 1`] = `
<span
  aria-hidden={true}
  className="fa fa-home sc-cSHVUG knnGKx sc-gZMcBi hbkhKP"
/>
`;

exports[`Icon Element applies right modifier 1`] = `
<span
  aria-hidden={true}
  className="fa fa-home sc-cSHVUG kbtYbk sc-gZMcBi hbkhKP"
/>
`;

exports[`Icon Element renders correctly 1`] = `
<span
  aria-hidden={true}
  className="fa fa-home sc-cSHVUG isCEZI sc-gZMcBi hbkhKP"
/>
`;
